py/desispec/validredshifts.py

```diff
@@ -0,0 +1,152 @@
+# Example usage:
+# redrock_path = '/global/cfs/cdirs/desi/spectro/redux/guadalupe/tiles/cumulative/1392/20210517/redrock-5-1392-thru20210517.fits'
+# cat = validate(redrock_path, return_target_columns=True)
+
+import os, warnings
+import numpy as np
+from astropy.table import Table, hstack, join
+import fitsio
+
+
+def get_good_fiberstatus(cat, isqso=False):
+    if not isqso:
+        good_fiberstatus = cat['COADD_FIBERSTATUS']==0
+    else:
+        good_fiberstatus = (cat['COADD_FIBERSTATUS']==0) | (cat['COADD_FIBERSTATUS']==8388608) | (cat['COADD_FIBERSTATUS']==16777216)
+    good_fiberstatus &= cat['ZWARN'] & 2**9 == 0  # NO DATA flag
+    return good_fiberstatus
+
+
+def validate(redrock_path, fiberstatus_cut=True, return_target_columns=False, extra_columns=None, emline_path=None, ignore_emline=False):
+    '''
+    Validate the redshift quality with tracer-dependent criteria for redrock+afterburner results.
+
+    Args:
+       redrock_path: str, path of redrock FITS file
+
+    Options:
+       fiberstatus_cut: bool (default True), if True, impose requirements on COADD_FIBERSTATUS and ZWARN
+       return_target_columns: bool (default False), if True, include columns that indicate if the object belongs to each class of DESI targets
+       extra_columns: list of str (default None), additional columns to include in the output
+       emline_path: str (default None), specify the location of the emline file; by default the emline file is in the same directory as the redrock file
+       ignore_emline: bool (default False), if True, ignore the emline file and do not validate the ELG redshift
+    '''
+
+    output_columns = ['good_bgs', 'good_lrg', 'good_elg', 'good_qso']
```

Comment:
elsewhere throughout the code we use UPPERCASE column names, and here if you include `return_target_columns=True` you end up with a table with a mix of UPPER and lowercase.  Let's use UPPERCASE everywhere, even if it is a bit ugly retro.